Assets/MixedRealityToolkit.Services/InputAnimation/IMixedRealityInputRecordingService.cs

```diff
@@ -0,0 +1,70 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using UnityEngine;
+
+namespace Microsoft.MixedReality.Toolkit.Input
+{
+    /// <summary>
+    /// Provides input recording into an internal buffer and exporting to files.
+    /// </summary>
+    public interface IMixedRealityInputRecordingService : IMixedRealityInputDeviceManager
+    {
+        /// <summary>
+        /// Input is being recorded.
+        /// </summary>
+        bool IsRecording { get; }
+
+        /// <summary>
+        /// Limit the size of the recording buffer.
+        /// </summary>
+        /// <remarks>
+        /// If recording is limited any input older than the RecordingBufferTimeLimit will be discarded.
+        /// </remarks>
+        bool UseBufferTimeLimit { get; set; }
+
+        /// <summary>
+        /// Size of the input recording buffer.
+        /// </summary>
+        float RecordingBufferTimeLimit { get; set; }
+
+        /// <summary>
+        /// Start unlimited input recording.
+        /// </summary>
+        void StartRecording();
+
+        /// <summary>
+        /// Stop recording input.
+        /// </summary>
+        void StopRecording();
+
+        /// <summary>
+        /// Discard all recorded input
+        /// </summary>
+        void DiscardRecordedInput();
+
+        /// <summary>
+        /// Save recorded input animation to a file.
+        /// </summary>
+        /// <param name="directory">Directory in which to create the file. If null the persistent data path of the app is used.</param>
+        /// <returns>File path where input has been recorded.</returns>
+        /// <remarks>
+        /// Filename is determined automatically.
+        /// </remarks>
+        string SaveInputAnimation(string directory = null);
+
+        /// <summary>
+        /// Save recorded input animation to a file.
+        /// </summary>
+        /// <param name="filename">Name of the file to create.</param>
+        /// <param name="directory">Directory in which to create the file. If null the persistent data path of the app is used.</param>
+        /// <returns>File path where input has been recorded.</returns>
+        string SaveInputAnimation(string filename, string directory = null);
+
+        /// <summary>
+        /// Generate a file name for export.
+        /// </summary>
+        string GenerateOutputFilename();
```

Comment:
I haven't read on quite yet, but looking at this function and the ones above it, it almost seems like this is an implementation detail of how SaveInputAnimation(directory) works. Am I reading this correctly in that, in there were only a single overload (SaveInputAnimatino(filename, directory)), GenerateOutputFilename wouldn't be needed?

Reply:
I've moved this into the static serialization utils now, and just pass a base name and appendTimestamp flag parameters instead of using the profile.

In this PR the function seems a bit pointless, but in [the follow-up PR to add editor recording tools](https://github.com/lukastoenneMS/MixedRealityToolkit-Unity/pull/2/files) it's used to initialize the suggested filename for export file selection dialogs.